### PR TITLE
refactor: rewrite is_valid_port()

### PR DESCRIPTION
**Pull request by murrant**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Take two.  Fixed the error and added tests.
fix: `return $default;` -> `return self::get($key, $default);`

---
This change is [Reviewable](https://reviewable.io/reviews/librenms/librenms/7360)

### CODE REVIEW

**Comment by laf** [comment]
Tested fine for me.

Is it worth adding something like: https://p.libren.ms/view/raw/b62397e4 (`tests/PortsTest.php`)

**Comment by murrant** [comment]
TLDR: yes

Higher level tests are almost universally better.  They test more and the right things.  Also allow for refactoring without re-writing tests.

We can't go all the way to the top level with librenms tests right now, so, we build up from the bottom.

**Comment by network-guy** [comment]
LGTM

**Comment by laf** [comment]
So adding it in or not for now @murrant as the PR itself is good to go.

**Comment by murrant** [comment]
@laf This is good to merge for me.  We can wait to add the higher level tests I talked about separately.
